### PR TITLE
integration: Build local-gadget-test in the same container as local-gadget

### PR DESCRIPTION
**Pull request by burak-ok**
# integration: Build local-gadget-test in the same container as local-gadget

Since i am on `Ubuntu 22.04` i get the following error when i try to run the integration tests for `local-gadget`:
```bash
.....
/home/burak/codes/inspektor-gadget/integration/local-gadget/bin/minikube-v1.27.0 -p ${MINIKUBE_PROFILE} ssh "sudo local-gadget-integration.test -test.v -integration -container-runtime docker ${INTEGRATION_TESTS_PARAMS}"
Checking minikube with profile minikube-docker is running ...
Preparing minikube with profile minikube-docker for testing ...
Running test in minikube with profile minikube-docker ...
local-gadget-integration.test: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by local-gadget-integration.test)
local-gadget-integration.test: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by local-gadget-integration.test)
ssh: Process exited with status 1
make: *** [Makefile:70: test-docker] Error 1
make: Leaving directory '/home/burak/codes/inspektor-gadget/integration/local-gadget'
```

In order to have the same dependencies in `local-gadget` and `local-gadget-integration.test` i have moved the compilation into a similar Docker container.
Negative effect: `go test -c ...` directly on the host is 20 seconds faster on my machine. When there are no changes the waiting time is negligible.

Therefore the question:
Do we want to create 2 different targets? One with compilation on host and one in the Docker container?

### CODE REVIEW

**Comment by mqasimsarfraz** [comment]
Interesting, I was wondering if we could avoid having duplicate `Dockerfiles`. Just as an idea having a single builder image and specifying `workdir` and `entrypoint` at runtime. What do you say?

**Comment by burak-ok** [comment]
That would be even better. Caching would also work that way for all steps which use that container

**Comment by burak-ok** [comment]
Thought a little bit about it further. The `.dockerignore` files are different for each build.
Now if a file in `integration/..` gets changed the Container for `local-gadget` or `kubectl-gadget` would get recreated and a new compilation happens.

File: Dockerfiles/local-gadget-tests.Dockerfile
Comment (mqasimsarfraz) [inline Dockerfiles/local-gadget-tests.Dockerfile:14]: How about moving them to a common builder image (say `lg-builder`)? both the images can still use `lg-builder` and have independent `dockerignorefiles`. (maybe [`ONBUILD`](https://docs.docker.com/engine/reference/builder/#onbuild) in useful for `COPY/RUN/ADD`). Do you think it will be useful?

Comment (burak-ok) [reply Dockerfiles/local-gadget-tests.Dockerfile:14]: Did you mean it in this way?

Comment (mqasimsarfraz) [reply Dockerfiles/local-gadget-tests.Dockerfile:14]: Yep. This is looking better to me. These lines can be removed from here I reckon?

File: Dockerfiles/local-gadget-tests.Dockerfile
Comment (mqasimsarfraz) [inline Dockerfiles/local-gadget-tests.Dockerfile:1]: nit: `AS builder` can be removed.

File: Makefile
Comment (mqasimsarfraz) [inline Makefile:86]: we don't need this step anymore.

File: integration/local-gadget/Makefile
Comment (mqasimsarfraz) [inline integration/local-gadget/Makefile:79]: shouldn't we keep this? or am i missing something?

Comment (burak-ok) [reply integration/local-gadget/Makefile:79]: I don't why we would or want to remove the text executable.
If it is there we don't need to build or copy it

**Review by mqasimsarfraz — APPROVED** [review]
Looking good! Seems to be working fine for me (Although I noticed sometime the build was taking bit longer maybe because of go mod cache?)

Let's wait for another approval from the team before we let this in!

File: Dockerfiles/local-gadget-tests.Dockerfile.dockerignore
Comment (blanquicet) [inline Dockerfiles/local-gadget-tests.Dockerfile.dockerignore:5]: Is tools folder really needed?

File: integration/local-gadget/Makefile
Comment (blanquicet) [reply integration/local-gadget/Makefile:79]: Mmmmm unless I am missing something, there is no other way to be sure that the existing `local-gadget-integration.test` is using the latest code than building `local-gadget-tests` again, and it will overwrite `local-gadget-integration.test`. Therefore, I don't see the advantage of keeping it.

**Review by blanquicet — APPROVED** [review]
LGTM!

File: integration/local-gadget/Makefile
Comment (burak-ok) [reply integration/local-gadget/Makefile:79]: Hmmm, makes sense.
There is no Makefile or other relationship in the build chain from `local-gadget-integration.test` to the source files

File: Dockerfiles/local-gadget-tests.Dockerfile.dockerignore
Comment (burak-ok) [reply Dockerfiles/local-gadget-tests.Dockerfile.dockerignore:5]: Nope, removed it

**Pull request merged**
